desc/grid.py

```diff
@@ -1617,3 +1787,65 @@ def find_least_rational_surfaces(
     io = find_most_distant(io_rat, n, a, b, tol=atol, **kwargs)
     rho = _find_rho(iota, io, tol=atol)
     return rho, io
+
+
+def _periodic_spacing(x, period=2 * jnp.pi, sort=False, jnp=jnp):
```

Comment:
any reason for making `jnp` an argument?

Reply:
If I just use jax.numpy for these functions, then the nodes and spacing of linear grid also become jax arrays. If i recall correctly, tihs caused an error with one of current potential classes that made a lineargrid inside a traced fori_loop,

Reply:
That error might be fixed with #1002 if it was related to `compute_magnetic_field` but not sure